syntax = "proto3";
package vire.cms;

import "vire/utility/ModelIdentifier.proto";
import "vire/utility/InvalidContextError.proto";
import "vire/cms/ResourceBaseResponse.proto";
import "vire/cms/InvalidResourceError.proto";
import "vire/cms/InvalidStatusError.proto";
import "vire/cms/ArgumentError.proto";
import "vire/cms/ResourceExecError.proto";

message ResourceExecFailureResponse {

  vire.cms.ResourceBaseResponse _base = 1;

  vire.utility.ModelIdentifier error_type_id = 100;
  oneof error_oneof {
    vire.utility.InvalidContextError invalid_context_err  = 101;
    vire.cms.InvalidResourceError    invalid_resource_err = 102;
    vire.cms.InvalidStatusError      invalid_status_err   = 103;
    vire.cms.ArgumentError           argument_err         = 104;
    vire.cms.ResourceExecError       exec_err             = 105;
  }

}